plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -428,21 +459,23 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     screenCoordinate: ScreenCoordinate,
     callback: QueryAnnotationCallback<T>
   ) {
-    mapFeatureQueryDelegate.queryRenderedFeatures(
-      screenCoordinate,
-      RenderedQueryOptions(
-        listOf(
-          layer.layerId
-        ),
-        literal(true)
-      )
-    ) { features ->
-      features.value?.let { featureList ->
-        if (featureList.isNotEmpty()) {
-          val id = featureList.first().getProperty(getAnnotationIdKey()).asLong
-          callback.onQueryAnnotation(annotations[id])
+    if (layer != null) {
+      mapFeatureQueryDelegate.queryRenderedFeatures(
+        screenCoordinate,
+        RenderedQueryOptions(
+          listOf(layer!!.layerId),
+          literal(true)
+        )
+      ) { features ->
+        features.value?.let { featureList ->
```

Comment:
If `feature.value == null` - we will not call `callback.onQueryAnnotation(null)`. Is it expected behavior?

Reply:
Yes, everywhere get the result will check null. Have thought about it, it seems we don't need the callback interface to return a nullable result.